automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala

```diff
@@ -174,49 +175,51 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
     await condition isKernelShutdown
   }
 
+  /**
+    * Throw TimeoutException if Kernel is not ready after restart when timeout is reached
+    *
+    * @param timeout
+    */
   def restartKernel(timeout: FiniteDuration = 1 minute): Unit = {
     click on kernelMenu
     click on (await enabled restartKernelSelection)
     click on (await enabled restartKernelConfirmationSelection)
-    waitForIdleKernel(timeout)
-  }
-
-  /**
-    * wait for kernel icon to become idle. default timeout == 1.minute
-    */
-  def waitForIdleKernel(timeout: FiniteDuration = 1.minutes): Unit = {
-    await condition (
-      !cellsAreRunning && isKernelIdle() &&
-        find(id("notification_kernel")).exists(_.underlying.getCssValue("display") == "none"), timeout.toSeconds
-    )
-    await condition (find(id("kernel_indicator_icon")).exists(_.isDisplayed), 2) // little time to check isDisplayed to prevent StaleWebElementException
+    await notVisible restartKernelConfirmationSelection
+    await condition (isKernelReady && kernelNotificationText() == "none", timeout.toSeconds)
   }
 
   /**
     * wait for kernel to become ready. default timeout == 1.minute
+    *
+    * Throw TimeoutException if Kernel is not ready when timeout is reached
     */
-  def awaitReadyKernel(timeout: FiniteDuration = 1.minute): Unit = {
-    try {
-      waitForIdleKernel(timeout)
-    } catch {
-      case _: TimeoutException =>
-        if (isKernelDisconnected()) {
-          restartKernel(timeout)
-        }
+  def waitReadyKernel(timeout: FiniteDuration = 1.minutes): Unit = {
+    Try {
```

Comment:
Note that the `Try-recover` below is no different than `try-catch` in that it doesn't help with function composition. Again, not a blocker for this PR and I'd be happy to talk in person if interested.

As another note, `waitReadyKernel` could be rewritten to accept a function such that the `open` method could turn into something like

```scala
override def open(implicit webDriver: WebDriver): NotebookPage = {
    withReadyKernel {
        _ => super.open.asInstanceOf[NotebookPage]
    }
}
```
so methods like `open` wouldn't have to do any explicit waiting and kernel waiting related nitty gritty could be hidden in one place...similarly to the pattern we use with `withBillingProject`, `withNewNotebookPage`, etc.